src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -441,12 +441,24 @@ private static int GetMediaTypeLength(string input, int startIndex, out MediaTyp
             return current - startIndex;
         }
 
-        private static int GetMediaTypeExpressionLength(string input, int startIndex, out string mediaType)
+        internal static int ParseMediaTypeExpresion<T, K>(
+            string input,
+            int startIndex,
+            out K mediaType,
+            Func<string, int, int, T> typeAndSubtypeFactory,
```

Comment:
Would a comment suffice? I don't want to add an interface for something internal, I think it's a bit overkill having to implement a whole class and create an instance for the class just to consume the method, on the other side, if you use lambdas the compiler is smart about it and doesn't impact memory allocations as far as I've measured.